checks_windows.go

```diff
@@ -0,0 +1,410 @@
+}
+
+func (c cond) RegistryKey() (bool, error) {
+	registryArgs := regexp.MustCompile(`[\\]+`).Split(c.Key, -1)
+	if len(registryArgs) < 2 {
+		fail("Invalid key for RegistryKey. Did you supply 'key'?")
+		return false, errors.New("invalid registry key path: " + c.Key)
+	}
+	registryHiveText := registryArgs[0]
+	keyPath := fmt.Sprintf(strings.Join(registryArgs[1:len(registryArgs)-1], `\`))
+	keyLoc := registryArgs[len(registryArgs)-1]
+
+	var registryHive registry.Key
+	switch registryHiveText {
+	case "HKEY_CLASSES_ROOT", "HKCR":
+		registryHive = registry.CLASSES_ROOT
+	case "HKEY_CURRENT_USER", "HKCU":
+		registryHive = registry.CURRENT_USER
+	case "HKEY_LOCAL_MACHINE", "HKLM", "MACHINE":
+		registryHive = registry.LOCAL_MACHINE
+	case "HKEY_USERS", "HKU":
+		registryHive = registry.USERS
+	case "HKEY_CURRENT_CONFIG", "HKCC":
+		registryHive = registry.CURRENT_CONFIG
+	case "SOFTWARE":
+		registryHive = registry.LOCAL_MACHINE
+		keyPath = `SOFTWARE\` + keyPath
+	default:
+		fail("Unknown registry hive: " + registryHiveText)
+		return false, errors.New("Unknown registry hive " + registryHiveText)
+	}
+
+	debug("Getting key", keyPath, "from hive ", registryHiveText)
+	// Actually get the key
+	k, err := registry.OpenKey(registryHive, keyPath, registry.QUERY_VALUE)
+	if err != nil {
+		if verboseEnabled {
+			warn("Registry opening key failed:", err)
+		}
+		return false, err
+	}
+	defer k.Close()
+
+	// Fetch registry value
+	registrySlice := make([]byte, 256)
+	regLength, valType, err := k.GetValue(keyLoc, registrySlice)
+	if err != nil {
+		// Error is probably about the key not existing. This is fine, some keys
+		// are not defined until the setting is explicitly set. However, the
+		// check should not pass for RegistryKey or RegistryKeyNot, so we return
+		// an error.
+		warn("Failed to open registry key:", err)
+		return false, err
+	}
+
+	registrySlice = registrySlice[:regLength]
+	debug("Retrieved registry value was", registrySlice, "length", regLength, "value", valType)
+
+	// Determine value type to convert to string
+	var registryValue string
+	switch valType {
+	case 1: // SZ
+		registryValue, _, err = k.GetStringValue(keyLoc)
+	case 2: // EXPAND_SZ
+		registryValue, _, err = k.GetStringValue(keyLoc)
+	case 3: // BINARY
+		fail("Binary registry format not yet supported.")
+	case 4: // DWORD
+		registryValue = strconv.FormatUint(uint64(binary.LittleEndian.Uint32(registrySlice)), 10)
+	default:
+		fail("Unknown registry type: " + fmt.Sprint(valType))
+	}
+
+	// fmt.Printf("Registry value: %s, keyvalue %s\n", registryValue, keyValue)
+	if registryValue == c.Value {
+		return true, err
+	}
+	return false, err
+}
+
+func (c cond) RegistryKeyExists() (bool, error) {
+	_, err := c.RegistryKey()
+	if err != nil {
+		if err == registry.ErrNotExist {
+			return false, nil
+		} else {
+			return false, err
+		}
+	} else {
+		return true, nil
+	}
+}
+
+func (c cond) ScheduledTaskExists() (bool, error) {
```

Comment:
Should this be a new issue? Having as little `CommandOutput` content in the SE seems like the smartest option

Reply:
Good idea, but let's add a patch for this in a future PR so we get this merged as soon as possible

Reply:
I definitely will make it a different PR, but I was just checking.